src/sortStudents.ts

```diff
@@ -1,16 +1,51 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortOrder: number = order === 'asc' ? 1 : -1;
+  const copyStudents = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  function findAverageGrade(grades: number[]): number {
+    return grades.reduce((sum, curr) => sum + curr) / grades.length;
+  }
+
+  copyStudents.sort((a, b) => {
+    switch (sortBy) {
+      case SortType.Name:
+        return sortOrder * a.name.localeCompare(b.name);
+      case SortType.Surname:
+        return sortOrder * a.surname.localeCompare(b.surname);
```

Comment:
[CODE STYLE] - DRY. Combine logic for sorting by fields with same data types.
```suggestion
      case SortType.Name:
      case SortType.Surname:
        return sortOrder * a[sortBy].localeCompare(b[sortBy]);
```

Same for `Age` and `Married`